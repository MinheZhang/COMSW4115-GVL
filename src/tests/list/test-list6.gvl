// list_iterator

int main() {
  list l;
  l = create_list();

  node n1;
  n1 = create_node(0.0, 1.0, 1.0, 255, 255, 255, 0);
  node n2;
  n2 = create_node(10.0, 11.0, 1.0, 255, 255, 255, 0);
  node n3;
  n3 = create_node(20.0, 21.0, 1.0, 255, 255, 255, 0);
  node n4;
  n4 = create_node(30.0, 31.0, 1.0, 255, 255, 255, 0);
  node n5;
  n5 = create_node(40.0, 41.0, 1.0, 255, 255, 255, 0);

  insert_front(l, n1);
  insert_front(l, n2);
  insert_front(l, n3);
  insert_front(l, n4);
  insert_front(l, n5);
  
  list_iterator iter;
  for (iter = list_begin(l); iter != list_end(); iter = list_iter_next(iter)) {
    node n;
    n = list_iter_data(iter);
    printf(get_node_y(n));
  }

  node n_rm;
  n_rm = remove_front(l);
  printf(get_node_y(n_rm));
  destroy_node(n_rm);

  destroy_list(l);
}