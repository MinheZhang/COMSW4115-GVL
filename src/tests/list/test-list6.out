41
31
21
11
1
41